git_aggregator/repo.py

```diff
@@ -192,9 +192,45 @@ def aggregate(self):
         self._execute_shell_command_after()
         logger.info('End aggregation of %s', self.cwd)
 
+    def _use_autoshare(self):
+        """Check wether `git-autoshare` is available or not"""
+        status, result = subprocess.getstatusoutput("git autoshare-clone")
+        if status == 1:
+            return False
+        logger.info("Git autoshare is installed. Taking advantage of it...")
+        return True
+
     def init_repository(self, target_dir):
-        logger.info('Init empty git repository in %s', target_dir)
-        self.log_call(['git', 'init', target_dir])
+        repository = None
+        for remote in self.remotes:
+            # If a target is defined, use it as the base repository
+            # If not, choose the first merge
+            if remote["name"] == self.target["remote"]:
+                repository = remote["url"]
+                break
+            elif remote["name"] == self.merges[0]["remote"]:
+                repository = remote["url"]
+        branch = self.target["branch"]
+        logger.info(
+            'Cloning git repository %s (branch %s) in %s',
+            repository,
+            branch,
+            target_dir,
+        )
+        cmd = ('git',)
+        if self._use_autoshare():
+            # Start with a prefetch to update the cache and then clone
+            logger.info('Updating cache for %s', repository)
```

Comment:
I think it's good to tell which cache you are referring to.

```suggestion
            logger.info('Updating autoshare cache for %s', repository)
```